### PR TITLE
Rename `--tests` option to `--skip-tests` in Quarkus CLI

### LINKED ISSUE #18391: Quarkus CLI test are run even if `--no-tests` is specified
## Describe the bug
When building using the CLI with command `quarkus build --no-tests`  the tests are still executed.

### Expected behavior
No tests should be run

### Actual behavior
Tests are run 

## To Reproduce

Link to a small reproducer (preferably a Maven project if the issue is not Gradle-specific).

Or attach an archive containing the reproducer to the issue.

1.  `quarkus create app -a bug-reproducer` 
2.  `cd bug-reproducer`
3.  `quarkus build --no-tests`

### Screenshots
(If applicable, add screenshots to help explain your problem.)

### Output of `uname -a` or `ver`
Darwin tqvarnst-mac 20.5.0 Darwin Kernel Version 20.5.0: Sat May  8 05:10:33 PDT 2021; root:xnu-7195.121.3~9/RELEASE_X86_64 x86_64

### Output of `java -version`
openjdk version "16.0.1" 2021-04-20
OpenJDK Runtime Environment AdoptOpenJDK-16.0.1+9 (build 16.0.1+9)
OpenJDK 64-Bit Server VM AdoptOpenJDK-16.0.1+9 (build 16.0.1+9, mixed mode, sharing)

### Quarkus version or git rev
2.0.0.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)
Quarkus CLI
```
> quarkus --version
Client Version 2.0.0.Final
```

## Additional context
(Add any other context about the problem here.)

### PR DESCRIPTION
**Pull request by tqvarnst**
Fix #18391

Currently, the CLI is documented to default run the test and then the user can control it with --tests or --no-tests. However, the way that Piccoli works a boolean command line option works adding the option will flip the default value, which means that when you add --tests it actually set io.quarkus.cli.common.BuildOptions.runTest to false, and the negotiated flag --no-tests set it to true given the inverted behavior of what was intended. 

In this pull request I suggest that we rename the parameter to `--skip-tests` remove the negotable option, which gives us the intended behavior where default the CLI build task will run tests, but you can turn if off with `--skip-tests`.

### CODE REVIEW

**Comment by tqvarnst** [comment]
@gsmet Adding you as a review since this is potentially a breaking change, but in reality, I don't think it will affect any users since the current `--no-tests` parameters don't work as intended. Better to have a quick fix before users start to depend on the fault `--tests` flag that actually turns off the testing.

File: devtools/cli/src/main/java/io/quarkus/cli/common/BuildOptions.java [hunk 1]
```diff
@@ -13,7 +13,7 @@
     @CommandLine.Option(order = 5, names = { "--offline" }, description = "Work offline.", defaultValue = "false")
     public boolean offline = false;
 
-    @CommandLine.Option(order = 6, names = { "--tests" }, description = "Run tests.", negatable = true)
+    @CommandLine.Option(order = 6, names = { "--skip-tests" }, description = "Skip tests.")
     public boolean runTests = true;
```
Comment (gsmet) [inline devtools/cli/src/main/java/io/quarkus/cli/common/BuildOptions.java:17]: Shouldn't the meaning of this property be changed too? I suspect it won't work as is (or I might miss how Picocli is working as I have no experience with it).

**Review by gsmet — CHANGES_REQUESTED** [review]

**Comment by gsmet** [comment]
This will require a proper rebase to avoid the merge commit before merging.

**Comment by ebullient** [comment]
i don't agree with this. I would like to fix --tests and --no-tests.

File: devtools/cli/src/main/java/io/quarkus/cli/common/BuildOptions.java (hunk 1 above)
Comment (ebullient) [reply devtools/cli/src/main/java/io/quarkus/cli/common/BuildOptions.java:17]: I don't agree with this change. I am on PTO, but I would prefer to make the options as they exist work (which I thought I had tests for, but apparently not...).

**Review by ebullient — CHANGES_REQUESTED** [review]
Let's fix behavior of existing options. Picocli can do this just fine, I just missed a validating step along the way

**Comment by gsmet** [comment]
I'm closing this one as apparently this is not what we want. @ebullient can you take care of the follow-up? Thanks!

**Comment by ebullient** [comment]
back today.. yes, I will

**Pull request closed without merging**